File: src/LudoKing.Shared/Constants/BoardConstants.cs
namespace LudoKing.Shared.Constants;

public static class BoardConstants
{
    public const int OuterTrackLength = 52;
    public const int HomeStretchLength = 5;
    public const int HomeCenterSquare = 999;
    public const int YardSquare = -1;

    public static readonly string[] Colors = { "Red", "Blue", "Green", "Yellow" };

    public static readonly int[] SafeSquares = { 0, 8, 13, 21, 26, 34, 39, 47 };

    public static readonly Dictionary<string, int> ColorStartSquares = new()
    {
        { "Red",    0  },
        { "Blue",   13 },
        { "Green",  26 },
        { "Yellow", 39 }
    };

    public static readonly Dictionary<string, int> ColorHomeEntrySquares = new()
    {
        { "Red",    50 },
        { "Blue",   11 },
        { "Green",  24 },
        { "Yellow", 37 }
    };

    public static readonly Dictionary<string, int> HomeStretchStart = new()
    {
        { "Red",    100 },
        { "Blue",   110 },
        { "Green",  120 },
        { "Yellow", 130 }
    };

    public static bool IsSafeSquare(int square)
    {
        if (IsHomeStretchSquare(square)) return true;
        foreach (var s in SafeSquares)
            if (s == square) return true;
        return false;
    }

    public static bool IsHomeStretchSquare(int square) =>
        (square >= 100 && square <= 104) ||
        (square >= 110 && square <= 114) ||
        (square >= 120 && square <= 124) ||
        (square >= 130 && square <= 134);

    public static int GetHomeStretchStart(string color) => HomeStretchStart[color];

    public static int GetStartSquare(string color) => ColorStartSquares[color];

    public static int GetHomeEntrySquare(string color) => ColorHomeEntrySquares[color];
}

File: src/LudoKing.UI/src/app/shared/constants/board-constants.ts
// Must stay in sync with LudoKing.Shared/Constants/BoardConstants.cs

export const OUTER_TRACK_LENGTH = 52;
export const HOME_STRETCH_LENGTH = 5;
export const HOME_CENTER_SQUARE = 999;
export const YARD_SQUARE = -1;

export const COLOR_START_SQUARES: Record<string, number> = {
  Red: 0, Blue: 13, Green: 26, Yellow: 39
};

export const COLOR_HOME_ENTRY_SQUARES: Record<string, number> = {
  Red: 50, Blue: 11, Green: 24, Yellow: 37
};

export const HOME_STRETCH_START: Record<string, number> = {
  Red: 100, Blue: 110, Green: 120, Yellow: 130
};

export const SAFE_SQUARES = new Set([0, 8, 13, 21, 26, 34, 39, 47]);

export function isHomeStretchSquare(square: number): boolean {
  return square >= 100 && square <= 134;
}

export function isSafeSquare(square: number): boolean {
  return SAFE_SQUARES.has(square);
}

export const PLAYER_COLORS = ['Red', 'Blue', 'Green', 'Yellow'];

export const COLOR_HEX: Record<string, string> = {
  Red: '#e53935', Blue: '#1e88e5', Green: '#43a047', Yellow: '#fdd835'
};

// 15×15 board pixel helpers (canvas 600×600)
export const BOARD_SIZE = 600;
export const CELL = BOARD_SIZE / 15;  // 40px per cell
